libvast/test/bloom_filter.cpp

```diff
@@ -210,3 +210,16 @@ TEST(bloom filter - simple hasher and partitioning) {
   REQUIRE_EQUAL(err, caf::none);
   CHECK(x == y);
 }
+
+TEST(bloom filter - duplicate tracking) {
+  bloom_filter_parameters xs;
+  xs.m = 1_M;
+  xs.p = 0.1;
+  auto x = vast::test::unbox(
+    make_bloom_filter<xxhash, double_hasher, policy::no_partitioning,
+                      policy::duplicate_tracking>(xs));
```

Comment:
If you swap the two policy arguments you get very much undesired behavior. Let's make both enum classes instead with two arguments each.

```
enum class partition_bloom_filter {
  yes,
  no,
};

enum class track_duplicates {
  yes,
  no,
};
```

Reply:
I'm not aware of this idiom. So instead of passing an unconstrained type template, pass an enum?

I assume these should also be in namespace `policy`, right?

Reply:
What I like about this is the user side of things: 
```cpp
    if constexpr (tracking_policy == policy::tracking::no) {
      ...
    }
```

Much easier to read than the type-level comparison with `is_same`.